packages/flutter/lib/src/rendering/layer.dart

```diff
@@ -605,14 +655,17 @@ class ClipRectLayer extends ContainerLayer {
   ///
   /// The [clipRect] property must be non-null before the compositing phase of
   /// the pipeline.
-  ClipRectLayer({ this.clipRect });
+  ClipRectLayer({ this.clipRect, this.clipBehavior = Clip.antiAlias }) : assert(clipBehavior != null && clipBehavior != Clip.none);
```

Comment:
`assert(foo && bar)` is better written as `assert(foo), assert(bar)` because then when they fail, the error will be on the expression that failed, rather than one of two expressions that failed.